hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java

```diff
@@ -194,6 +194,18 @@ public ContainerInfo getContainer(final ContainerID containerID)
     return containerStateManager.getContainer(containerID);
   }
 
+  @Override
+  public boolean exists(ContainerID containerID) {
+    lock.lock();
```

Comment:
This should likely be filed as separate Jira, but why is this not a ReadWriteLock?

Reply:
Not sure. SCM ContainerManager has had a Reentrant lock from the first. I will investigate why it is not a RW lock, and create a new JIRA for this.